Onboarding/QA/tests/profile-pregnancy-health_test.exs

```diff
@@ -0,0 +1,438 @@
+      this_month_plus_eight_word = Calendar.strftime(this_month_plus_eight, "%B")
+
+      list_of_months = [
+        {"@datevalue(this_month, \"%B\")", "#{this_month_word}"},
+        {"@datevalue(this_month_plus_one, \"%B\")", "#{this_month_plus_one_word}"},
+        {"@datevalue(this_month_plus_two, \"%B\")", "#{this_month_plus_two_word}"},
+        {"@datevalue(this_month_plus_three, \"%B\")", "#{this_month_plus_three_word}"},
+        {"@datevalue(this_month_plus_four, \"%B\")", "#{this_month_plus_four_word}"},
+        {"@datevalue(this_month_plus_five, \"%B\")", "#{this_month_plus_five_word}"},
+        {"@datevalue(this_month_plus_six, \"%B\")", "#{this_month_plus_six_word}"},
+        {"@datevalue(this_month_plus_seven, \"%B\")", "#{this_month_plus_seven_word}"},
+        {"@datevalue(this_month_plus_eight, \"%B\")", "#{this_month_plus_eight_word}"},
+        {"I don't know", "I don't know"}
+      ]
+
+      edd_confirmation_text = "Thank you! Just confirming your estimated due date 🗓️\r\n\r\nAre you expecting the baby on *25 #{this_month_plus_one_word} #{Calendar.strftime(this_month_plus_one, "%Y")}*?"
+
+      edd_month = String.pad_leading("#{this_month_plus_one.month}", 2, "0")
+      full_edd = Calendar.strftime(this_month_plus_one, "%Y") <> "-" <> "#{edd_month}" <> "-25"
+
+      setup_flow()
+      |> FlowTester.set_contact_properties(%{"pregnancy_status" => "", "checkpoint" => "", "profile_completion" => "", "edd" => "", "pregnancy_sentiment" => ""}) # Pregnancy Information
+      |> FlowTester.set_contact_properties(%{"year_of_birth" => "1988", "province" => "Western Cape", "area_type" => ""}) # Basic Information
+      |> FlowTester.set_contact_properties(%{"relationship_status" => "", "education" => "", "socio_economic" => "", "other_children" => ""}) # Personal Information
+      |> FlowTester.set_contact_properties(%{}) # Daily Life
+      |> FlowTester.set_contact_properties(%{"gender" => "", "name" => "Lily"})
+      |> FlowTester.set_contact_properties(%{"opted_in" => "true"})
+      |> FlowTester.start()
+      |> receive_message(%{
+        text: "I've got a *lot* of information on pregnancy" <> _,
+        buttons: button_labels(["I'm pregnant", "Partner is pregnant", "Just curious"])
+      })
+      |> FlowTester.send(button_label: "I'm pregnant")
+      |> contact_matches(%{"gender" => "female", "pregnancy_status" => "im_pregnant", "checkpoint" => "pregnant_mom_profile", "profile_completion" => "0%"})
+      |> receive_message(%{
+        text: "👤 *Which month are you expecting your baby to be born?*" <> _,
+        list: {"Month", ^list_of_months}
+      })
+      |> FlowTester.send(button_label: this_month_plus_one_word)
+      |> receive_message(%{
+        text: "👤 *On what day of the month are you expecting the baby to be born?*\n\nType in a number between 1 and 31." <> _
+      })
+      |> FlowTester.send("25")
+      |> receive_message(%{
+        text: edd_confirmation_text,
```

Comment:
Variable references in a match pattern are treated as assignments. If you want to match on the value of the variable, you need a `^` prefix: `text: ^edd_confirmation_text`.

(There's a warning about this in test output for variables that already exist.)

Reply:
I had some weird errors because of this, but it seems to be working now.